### ISSUE TITLE
Relative paths handling

### ISSUE THREAD

**Issue by clementhervy** [waf/RunOnSave#8]
Hey there, very nice extension.

I think it would be even nicer if we could use relative paths to the solution root instead of, or in parallel to, fully qualified paths. Something like a `path_origin` option with values `absolute` (which would be equivalent to the current default) and `relative_to_the_solution_root`? Or for a simpler solution, more placeholders like `{relative_file}` (just an example, this naming is suboptimal).

This suggestion springs from my personal experience : I tried to use RunOnSave to apply ReSharper formatting on a saved file (kinda like CodeCleanup, either VS default or ReSharper), and the formatting tool requires that you pass the solution file as an argument.
So when I save the file `C:\projects\MySolution\path\to\file\being\saved\FileSaved.cs` with :
```
[*.cs]
command = jb
arguments = cleanupcode --profile="Built-in: Reformat Code" --include="{file}" MySolution.sln
```
I get the following error message :
`Unable to find target solution file in path C:\projects\MySolution\path\to\file\being\saved\MySolution.sln`.
... simply because the tool is unable to locate the solution file `MySolution.sln` at `C:\projects\MySolution\path\to\file\being\saved\` because it's actually at the solution root `C:\projects\MySolution\`.

The fix is simple enough, I just need to set the working folder to the solution root, which I think is the intended use for it. But this leads to another problem : the saved file is not found, because the tool requires either a relative path or a filename mask.  I can use `{filename}` of course, but then if there are several files with the same name they're all formatted, which introduces changes unrelated to the current development cycle.

Another benefit for such a change would be to facilitate the use of the extension for teams, since not all developers working on a project will store its code at the same full path.

Cheers, and keep up the good work!

**Comment by waf** [comment]
Hi @clementhervy, thanks for raising this issue! If we added the following two placeholders, this would solve your issue, right? Assuming directory contents of `C:\MySolution\MySolution.sln` and `C:\MySolution\MyProject\Program.cs`

  - `{solution_directory}` - The directory of the solution file (e.g. C:\MySolution)
  - `{file_in_solution}` - The file that was saved, relative to the solution_directory (e.g. MyProject\Program.cs)

You could then configure:

```ini
[*.cs]
command = jb
arguments = cleanupcode --profile="Built-in: Reformat Code" --include="{file_in_solution}" {solution_directory}\MySolution.sln
```

I think this should work for you, but please let me know if I've misunderstood the constraints. Thanks!

**Comment by clementhervy** [comment]
> Hi @clementhervy, thanks for raising this issue! If we added the following two placeholders, this would solve your issue, right? Assuming directory contents of `C:\MySolution\MySolution.sln` and `C:\MySolution\MyProject\Program.cs`
> 
> * `{solution_directory}` - The directory of the solution file (e.g. C:\MySolution)
> * `{file_in_solution}` - The file that was saved, relative to the solution_directory (e.g. MyProject\Program.cs)
> 
> You could then configure:
> 
> ```ini
> [*.cs]
> command = jb
> arguments = cleanupcode --profile="Built-in: Reformat Code" --include="{file_in_solution}" {solution_directory}\MySolution.sln
> ```
> 
> I think this should work for you, but please let me know if I've misunderstood the constraints. Thanks!

Hey! That would be perfect, thank you!

**Comment by waf** [comment]
I've released v1.4 -- please give it a try and let me know if you run into any problems!
